tests/all-test.php

```diff
@@ -103,30 +101,113 @@ public function setUp() {
 			'track_page_types'          => array( 'page' ),
 		);
 		update_option( 'parsely', $option_defaults );
-		self::$parsely_html = <<<PARSELYJS
-<script data-cfasync="false" id="parsely-cfg" data-parsely-site="blog.parsely.com" src="https://cdn.parsely.com/keys/blog.parsely.com/p.js"></script>
-PARSELYJS;
 	}
 
+	public function test_class_version() {
+		self::assertEquals( '2.4.1', \Parsely::VERSION );
+	}
+
+	public function test_constant_version() {
+		self::assertEquals( \Parsely::VERSION, PARSELY_VERSION );
+	}
+
+	public function test_cache_buster() {
+		self::assertEquals( PARSELY_VERSION, \Parsely::get_asset_cache_buster() );
+	}
+
+	public function test_plugin_url_constant() {
+		self::assertTrue( defined( 'PARSELY_PLUGIN_URL' ) && is_string( PARSELY_PLUGIN_URL ) && strlen( PARSELY_PLUGIN_URL ) > 0 );
+	}
 
 	/**
-	 * Test the parsely tag.
-	 *
-	 * @category   Function
-	 * @package    SampleTest
+	 * Test the default parsely script enqueues.
 	 */
-	public function test_parsely_tag() {
-		echo 'NOW IT BEGINS';
+	public function test_parsely_default_scripts() {
 		ob_start();
 		$post_array = $this->create_test_post_array();
 		$post       = $this->factory->post->create( $post_array );
 		$this->go_to( '/?p=' . $post );
-		echo esc_html( self::$parsely->insert_parsely_javascript() );
+		echo self::$parsely->insert_parsely_javascript();
+
+		$intermediate_output = ob_get_contents();
+		self::assertEquals(
+			'',
+			$intermediate_output,
+			'Failed to confirm scripts were not printed by insert_parsely_javascript()'
+		);
+
+		self::assertTrue(
+			wp_script_is( 'wp-parsely-api', 'registered' ),
+			'Failed to confirm API script was registered'
+		);
+		self::assertFalse(
+			wp_script_is( 'wp-parsely-api', 'enqueued' ),
+			'Failed to confirm API script was not enqueued'
+		);
+
+		self::assertTrue(
+			wp_script_is( 'wp-parsely-tracker', 'enqueued' ),
+			'Failed to confirm tracker script was enqueued'
+		);
+
+		wp_print_scripts();
 		$output = ob_get_clean();
-		echo esc_html( $output );
-		self::assertContains( self::$parsely_html, $output );
+
+		self::assertEquals(
+			"<script type='text/javascript' data-parsely-site=\"blog.parsely.com\" src='https://cdn.parsely.com/keys/http://blog.parsely.com/p.js?ver=" . PARSELY_VERSION . "' id=\"parsely-cfg\"></script>\n",
+			$output,
+			'Failed to confirm script tags were printed correctly'
+		);
 	}
 
+	/**
+	 * Test the parsely script enqueues when an api secret is set.
+	 */
+	public function test_parsely_api_enabled_scripts() {
+		$options = get_option( 'parsely' );
+		$options['api_secret'] = 'hunter2';
+		update_option( 'parsely', $options );
```

Comment:
Might want to consider moving this utility step into its own method:

```php
private set_api_secret( $api_secret ) {
	$options = get_option( 'parsely' );
	$options['api_secret'] = $api_secret;
	update_option( 'parsely', $options );
}
```

Then:
```suggestion
		$this->set_api_password( 'hunter2' );
```

Reply:
Or if the other tests set other settings, then make it slightly more generic:

```php
private update_parsely_option( $key, $value ) {
	$options = get_option( 'parsely' );
	$options[ $key ] = $value;
	update_option( 'parsely', $options );
}